open-craft/openedx-events-sender
# feat: initial implementation

**Description:** This implements the plugin which sends Open edX events as `POST` requests.

**JIRA:** [BB-6663](https://tasks.opencraft.com/browse/BB-6663)

**Dependencies:** None

**Merge deadline:** None

**Installation instructions:** See the README.

**Testing instructions:**

1. Install this plugin.
2. Go to `lms/envs/private.py` and set the following:
   ```python
   EVENT_SENDER_ENROLLMENT_URL = "https://webhook.site/xyz"  # Take this from https://webhook.site
   EVENT_SENDER_ENROLLMENT_HEADERS = {"Authorization": "TEST DATA"}
   ```
3. Enroll/unenroll a user in a course.
4. Check the POSTed data.
5. Set custom field mapping like this:
   ```python
   EVENT_SENDER_ENROLLMENT_FIELD_MAPPING = {
       "user_pii_email": "email",
       "course_course_key": "course_id",
       "is_active": "is_active"
   }
   ```
6. Verify that only these fields are passed in the POST. Their names should match new mappings.

**Reviewers:**

**Merge checklist:**

**Post merge:**
      finished.
